diff --git a/modules/custom/studiobridge_live_shoot_page/src/Form/StudioBridgeLiveShootingForm.php b/modules/custom/studiobridge_live_shoot_page/src/Form/StudioBridgeLiveShootingForm.php
index 7bde1bc..9ea53e9 100644
--- a/modules/custom/studiobridge_live_shoot_page/src/Form/StudioBridgeLiveShootingForm.php
+++ b/modules/custom/studiobridge_live_shoot_page/src/Form/StudioBridgeLiveShootingForm.php
@@ -441,6 +441,12 @@ public function productGetOrUpdateCallback(array &$form, FormStateInterface $for
       ->condition('bundle', 'sessions')
       ->execute()->fetchAll();
 
+      $sessions_nids_in_reshoot_field = $db->select('node__field_reshoot_product', 'c')
+        ->fields('c')
+        ->condition('field_reshoot_product_target_id', $new_or_old_product_nid)
+        ->condition('bundle', 'sessions')
+        ->execute()->fetchAll();
+
       // todo : if count is more than 1
       if (count($sessions_nids)) {
         // $session_id
@@ -451,6 +457,15 @@ public function productGetOrUpdateCallback(array &$form, FormStateInterface $for
           }
         }
       }
+      if (count($sessions_nids_in_reshoot_field)) {
+        // $session_id
+        foreach ($sessions_nids_in_reshoot_field as $field1) {
+          if ($field1->entity_id == $session_id) {
+            $reshoot = FALSE;
+            break;
+          }
+        }
+      }
       // If current product is reshoot then prompt user to confirm
       if ($reshoot && $_GET['identifier'] !== $identifier) {
         $inject_script = '<script>
